event_module.py

```diff
@@ -0,0 +1,33 @@
+import pygame
+import sys
+
+import coins_module
+import obstacles_module
+
+def setting_up_events():
+	# Generate coin once in every 1.5 seconds
+	pygame.time.set_timer(pygame.USEREVENT+1, 1500)
+	# Generate tree obstacles once in every 8 seconds
+	pygame.time.set_timer(pygame.USEREVENT+2, 8000)
+	# Generate other obstacles once in every 20 seconds
```

Comment:
This is bad programming! Obstacles have to be generated in Random. If a player knows he will face an obstacle every 20 seconds, then he will expect the obstacle beforehand rather than finding it unexpectedly!

Use random number generation with random() and provide a range!

Reply:
Yes senior. But this timer has to be preset before the game loop. Hence I had randomized the initial x position of the obstacle while creating the obstacle using the random_x variable.

`random_x = random.randint(background_module.bg.get_width(), background_module.bg.get_width()+1500) 
Tree.obstacles.append(Tree(random_x,random_num))`